### PR TITLE
install_test.go of error message amending

### PR DESCRIPTION
**Pull request by jianglingxia**
**What this PR does / why we need it**:
amend the error message
ref: https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/aws/tags_test.go#L91
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: federation/apis/federation/install/install_test.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func TestCodec(t *testing.T) {
 
 func TestInterfacesFor(t *testing.T) {
 	if _, err := api.Registry.GroupOrDie(federation.GroupName).InterfacesFor(federation.SchemeGroupVersion); err == nil {
-		t.Fatalf("unexpected non-error: %v", err)
+		t.Fatalf("expected an error: %v", err)
```
Comment (xiangpengzhao) [inline federation/apis/federation/install/install_test.go]: If so, what do you expect %v to output? note that `err==nil`.

**Comment by jianglingxia** [comment]
PTAL @irfanurrehman
/assign @mwielgus

**Comment by irfanurrehman** [comment]
/lgtm
/sig federation
cc @kubernetes/sig-federation-pr-reviews for approval

**Comment by marun** [comment]
/lgtm cancel 

As per @xiangpengzhao's comment, the message shouldn't be attempting to format a nil error message.

**Comment by irfanurrehman** [comment]
@marun My bad! My brain must have been absent from where its supposed to be when I half looked at it and passed for a simple string change. Thanks for having a look!
@jianglingxia please update the message as:
```
t.Fatalf("expected an error: got nil")
```

**Comment by xiangpengzhao** [comment]
@irfanurrehman would it be better to use `expected an error but got nil`?

:)

**Comment by irfanurrehman** [comment]
> @irfanurrehman would it be better to use expected an error but got nil?
> 
> :)

yes, or "expected an error, got nil" :)

**Comment by jianglingxia** [comment]
thanks very much ,I have updated it! @irfanurrehman @xiangpengzhao

File: federation/apis/federation/install/install_test.go [hunk 2]
```diff
@@ -68,7 +68,7 @@ func TestCodec(t *testing.T) {
 
 func TestInterfacesFor(t *testing.T) {
 	if _, err := api.Registry.GroupOrDie(federation.GroupName).InterfacesFor(federation.SchemeGroupVersion); err == nil {
-		t.Fatalf("unexpected non-error: %v", err)
+		t.Fatalf("expected an error,but got nil")
```
Comment (xiangpengzhao) [inline federation/apis/federation/install/install_test.go]: please add a space after the comma. 
and amend the commit message to be a more meaningful one.

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]

**Comment by jianglingxia** [comment]
PTAL @xiangpengzhao 
@mwielgus

**Comment by jianglingxia** [comment]
PTAL @irfanurrehman  @xiangpengzhao 
/assign @mwielgus

**Comment by jianglingxia** [comment]
/assign @mwielgus
PTAL thanks

**Comment by jianglingxia** [comment]
/assign @madhusudancs
PTAL

**Comment by jianglingxia** [comment]
please label 
/approve no-issue
thanks @quinton-hoole

**Comment by jianglingxia** [comment]
the pr has 2 months 
please label
/approve no-issue
thanks @quinton-hoole

**Comment by jianglingxia** [comment]
thanks @irfanurrehman  trouble you again label lgtm
before label but removed it by robot.
image

**Comment by irfanurrehman** [comment]
/lgtm cancel
@jianglingxia, please put a proper message in the commit log in place of  "squash the commits into one"
and I will lgtm again

**Comment by jianglingxia** [comment]
because of unknown repository,I can not update it in before branch,so I reopen the pr #50354,then I close the pr!thanks

**Pull request closed without merging**